src/environmentd/src/http.rs

```diff
@@ -453,6 +456,27 @@ impl InternalHttpServer {
     }
 }
 
+async fn internal_http_auth<B>(mut req: Request<B>, next: Next<B>) -> impl IntoResponse {
+    if let Some(user) = req.headers().get("x-materialize-user") {
```

Comment:
you could get rid of the if and else by doing `let user = req.headers().get("..").unwrap_or_else(|| SYSTEM_USER.clone())`